service/assessment/assessment.go

```diff
@@ -412,103 +422,53 @@ func (s *Service) initEvidenceStoreStream(URL string, additionalOpts ...grpc.Dia
 	evidenceStoreClient := evidence.NewEvidenceStoreClient(conn)
 	stream, err = evidenceStoreClient.StoreEvidences(context.Background())
 	if err != nil {
-		return nil, fmt.Errorf("could not set up stream for storing evidences: %w", err)
+		return nil, fmt.Errorf("could not set up stream to evidence store for storing evidences: %w", err)
 	}
 
+	log.Infof("Connected to Evidence Store")
+
 	return
 }
 
 // initOrchestratorStream initializes the stream to the Orchestrator
-func (s *Service) initOrchestratorStream() error {
+func (s *Service) initOrchestratorStream(URL string, additionalOpts ...grpc.DialOption) (stream orchestrator.Orchestrator_StoreAssessmentResultsClient, err error) {
 	log.Infof("Trying to establish a connection to orchestrator service @ %v", s.orchestratorAddress)
 
 	// Establish connection to orchestrator gRPC service
-	conn, err := grpc.Dial(s.orchestratorAddress,
-		api.DefaultGrpcDialOptions(s.orchestratorAddress, s, s.grpcOpts...)...,
-	)
+	err = s.initOrchestratorClient()
```

Comment:
Check first, if `s.orchestratorClient` is nil? Could be up already, e.g. through `Metrics(..)` function